crates/steel-core/src/steel_vm/primitives.rs

```diff
@@ -671,24 +674,51 @@ fn not(value: &SteelVal) -> bool {
     matches!(value, SteelVal::BoolV(false))
 }
 
+#[steel_derive::function(name = "number?", constant = true)]
+pub fn numberp(value: &SteelVal) -> bool {
+    matches!(
+        value,
+        SteelVal::IntV(_)
+            | SteelVal::BigNum(_)
+            | SteelVal::FractV(_)
+            | SteelVal::BigFract(_)
+            | SteelVal::NumV(_)
+    )
+}
+
 #[steel_derive::function(name = "int?", constant = true)]
 fn intp(value: &SteelVal) -> bool {
-    matches!(value, SteelVal::IntV(_))
+    matches!(value, SteelVal::IntV(_) | SteelVal::BigNum(_))
 }
 
 #[steel_derive::function(name = "integer?", constant = true)]
 fn integerp(value: &SteelVal) -> bool {
-    matches!(value, SteelVal::IntV(_))
+    intp(value)
 }
 
 #[steel_derive::function(name = "float?", constant = true)]
 fn floatp(value: &SteelVal) -> bool {
     matches!(value, SteelVal::NumV(_))
 }
 
-#[steel_derive::function(name = "number?", constant = true)]
-fn numberp(value: &SteelVal) -> bool {
-    matches!(value, SteelVal::NumV(_) | SteelVal::IntV(_))
+#[steel_derive::function(name = "real?", constant = true)]
```

Comment:
What's the preferred way of registering these functions? I tried to standardize on the `steel_derive` macros since I kind of like them.

Reply:
The derive macros are good! They're newer so some of the older functions don't have them. Doc comments on them automatically make it in to the docs so they're preferred when possible